KwartrackerTests/Mocks/MockAuthServiceClient.swift

```diff
@@ -0,0 +1,178 @@
+              "code": 422
+            }
+          ]
+        }
+        """
+    
+    private func createJSONObject(_ mockObject: String) -> JSONObject? {
+        guard let data = mockObject.data(using: .utf8) else { return nil }
+        
+        do {
+            if let jsonObject = try JSONSerialization.jsonObject(with: data,
+                                                                 options: .allowFragments) as? Dictionary<String, Any> {
+                return jsonObject
+            }
+        } catch let error {
+            print(error.localizedDescription)
+            return nil
+        }
+        
+        return nil
+    }
+}
+
+extension MockAuthServiceClient: AuthenticationServiceDelegate {
+    @discardableResult func signIn(
+        credentialInput input: CredentialsInput,
+        completion: @escaping (Result<GraphQLResult<SignInMutation.Data>, Error>) -> Void
+    ) -> Cancellable {
+        if shouldErrorRequest {
+            let error = NSError(domain: "", code: 422,
+                                userInfo: [NSLocalizedDescriptionKey: "Unable to make request"])
+            
+            completion(.failure(error))
+        } else {
+            if shouldLoginWithError {
+                if let errorJSONObject = createJSONObject(loginErrorResponse) {
+                    let error = GraphQLError(errorJSONObject)
+                    
+                    completion(.success(GraphQLResult<SignInMutation.Data>(
+                        data: nil,
+                        extensions: nil,
+                        errors: [error],
+                        source: .server,
+                        dependentKeys: Set<CacheKey>(arrayLiteral: CacheKey(0))
+                    )))
+                }
+            } else {
+                let data = SignInMutation.Data(signInWithEmail: loginResponse)
+                
+                completion(.success(GraphQLResult<SignInMutation.Data>(
+                    data: data,
+                    extensions: nil,
+                    errors: nil,
+                    source: .server,
+                    dependentKeys: nil
+                )))
+            }
+        }
+        
+        return MockServiceTransport()
+    }
+    
+    func signUp(
+        signUpCredential input: SignUpWithEmailInput,
+        completion: @escaping (Result<GraphQLResult<SignUpMutation.Data>, Error>) -> Void
+    ) -> Cancellable {
+        if shouldErrorRequest {
+            let error = NSError(domain: "", code: 422,
+                                userInfo: [NSLocalizedDescriptionKey: "Unable to make request"])
+            
+            completion(.failure(error))
+        } else {
+            if shouldSignupWithError {
+                if let errorJSONObject = createJSONObject(signupErrorResponse) {
+                    let error = GraphQLError(errorJSONObject)
+                    
+                    completion(.success(GraphQLResult<SignUpMutation.Data>(
```

Comment:
Since this is an error, better use 
```
completion(.failure(error))
```

Reply:
This is a mock error response from the API I just copied.